### PR TITLE
Use the QuarkusForkJoinWorkerThreadFactory for fork-join pools

### LINKED ISSUE #20067: Config property not found when in ForkJoin common Pool thread
### Describe the bug

Fetching a config property on ForkJoin common Pool thread failed.
Be aware that the behavior is not the same in dev mode and in prod mode as it's work in dev mode.

### Expected behavior

Same behavior in prod and dev mode is expected, cfg must be returned.

### Actual behavior

In prod mode, reading a property set in application.properties fails (return null).

### How to Reproduce?

For instance :

ThreadContext threadContext = ThreadContext.builder()
                .propagated(ThreadContext.ALL_REMAINING)
                .cleared(ThreadContext.TRANSACTION)
                .build();
        var expirationTask = threadContext.contextualRunnable(() -> {
           var cfg = ConfigProvider.getConfig().getOptionalValue("smallrye.jwt.sign.key.location", String.class).orElse(null);
        });
        CompletableFuture.runAsync(expirationTask, delayedExecutor(60, TimeUnit.SECONDS));
       

cfg is null.

### Output of `uname -a` or `ver`

 #36-Ubuntu SMP Thu Aug 26 19:22:09 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "11.0.11" 2021-04-20 OpenJDK Runtime Environment (build 11.0.11+9-Ubuntu-0ubuntu2) OpenJDK 64-Bit Server VM (build 11.0.11+9-Ubuntu-0ubuntu2, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.1.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Sanne**
An experiment to see if we can better control the Classloaders in the Fork/Join initiated threads.

@stuartwdouglas WDYT of the approach?  For dev-mode we'd likely want a slightly different implementation, so to be able to capture all threads when they are started, and then have the ability to swap the CL.

### CODE REVIEW

**Comment by radcortez** [comment]
Gave it a try with https://github.com/quarkusio/quarkus/issues/20067 and it seems to fix the issue. I think it makes sensed to move to this approach.

File: independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/forkjoin/QuarkusForkJoinWorkerThreadFactory.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package io.quarkus.bootstrap.forkjoin;
+
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinWorkerThread;
+
+public final class QuarkusForkJoinWorkerThreadFactory implements ForkJoinPool.ForkJoinWorkerThreadFactory {
```
Comment (radcortez) [inline independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/forkjoin/QuarkusForkJoinWorkerThreadFactory.java:6]: Maybe we want to apply this to `QuarkusDevModeTest`, `QuarkusUnitTest` and `QuarkusProdModeTest`?

Comment (Sanne) [reply independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/forkjoin/QuarkusForkJoinWorkerThreadFactory.java:6]: Probably? I'm not familiar with these

I do suspect thought that we'll want to have a slightly different implementation for "real prod mode" than for tests

**Comment by radcortez** [comment]
@Sanne @stuartwdouglas are we ok to move forward with this?

**Comment by Sanne** [comment]
Hi @radcortez - yes sorry I resumed yesterday with this.

I was trying to add some tests, but as I worked on it I realized my new integration tests were pointless, as it affects production mode only so @QuarkusTest isn't affected by the issue.

Do we have any existing pattern / utility to test such things? Any suggestion?

**Comment by radcortez** [comment]
Have a look here into how the `maven` integration tests work: https://github.com/quarkusio/quarkus/blob/main/test-framework/maven/src/main/java/io/quarkus/maven/it/RunAndCheckMojoTestBase.java#L21.

Not exactly what we need, because it runs `quarkus:dev`, but maybe it can be adjusted (or copied) to run in prod mode. I'm not sure if we have anything else.

I guess that it should be also possible to run the prod jar in `pre-integrations-tests` and then run tests with `failsafe`.

**Comment by Sanne** [comment]
thanks for the pointers!

**Review by radcortez — APPROVED** [review]

**Comment by radcortez** [comment]
Wondering if the test fails are related, but at first glance it doesn't seem like it.

**Review by stuartwdouglas — DISMISSED** [review]

**Comment by radcortez** [comment]
> @radcortez looks like we're ready to merge; is there any open issue this should be linked to?

I'm not aware of any other options issues. Please go ahead. Thanks!

**Pull request merged**